### PR TITLE
refine the indent

### PR DESCRIPTION
**Pull request by krizex**
Fix indent

### CODE REVIEW

**Comment by krizex** [comment]
@robhoes Hi Rob, I have refined the indent per your request, please take a look and merge when you get time.

**Comment by mseri** [comment]
It does not look better to me. Can you set-up your editor to show tabs as 4 spaces?

**Comment by mseri** [comment]
It does not look better to me. See, e.g., the `finally` on the first changed line. That should be indented 4 tabs, and instead they are 14. Please set-up your editor to show tabs as long as 4 spaces or 8 spaces and check the output

**Comment by krizex** [comment]
I see that, when I edit ocaml code I set tab equal to 2 spaces so the `tabify` action replace 2 spaces with one tab, I will update it to replace 4 spaces with 1 tab.

**Comment by krizex** [comment]
opam-indent does not support tab indent well, so I take a while to manually refine the code(delete some space after tabify).
I think the best way for indentation in ocaml is just use space, no tab.

**Comment by mseri** [comment]
I agree with you, however this module was already using tabs and unless we do an `ocp-indent` for the whole of it, we need to stick with the current style. In this case this means do it by hand (or use editors that support tabs and infer the styles like vscode or vim/emacs with opportune configuration)

File: xc/xenops_server_xen.ml [hunk 1]
```diff
@@ -1531,48 +1531,48 @@ module VM = struct
...
+						let k = vm.Vm.id in
+						let vmextra = DB.read_exn k in
+						let (build_info, timeoffset) = match vmextra.VmExtra.persistent with
+							| { VmExtra.build_info = None } ->
+								error "VM = %s; No stored build_info: cannot safely restore" vm.Vm.id;
+								raise (Does_not_exist("build_info", vm.Vm.id))
+							| { VmExtra.build_info = Some x; VmExtra.ty } ->
+								let initial_target = get_initial_target ~xs domid in
+								let timeoffset = match ty with
+										Some x -> (match x with HVM hvm_info -> hvm_info.timeoffset | _ -> "")
```
Comment (mseri) [inline xc/xenops_server_xen.ml]: Please use `| Some x` and unindent it of one tab

Comment (krizex) [inline xc/xenops_server_xen.ml]: Updated.

**Comment by krizex** [comment]
@mseri BTW, I use `ocp-indent-line` in emacs which provided by this layer(https://github.com/syl20bnr/spacemacs/tree/master/layers/%2Blang/ocaml) to reformat the code, then use `tabify` to replace spaces with tabs.
Can you tell me how to reformat the code with `ocp-indent` command line tool to indent the code with tab, so I can achieve my purpose faster?

**Comment by mseri** [comment]
Unfortunately `ocp-indent` does not work nicely with tabs, so we cannot use it in this case

**Comment by mseri** [comment]
It looks properly indented now.

As a general rule of thumbs, once we start nesting so many levels I tend to break down the code in multiple functions and reduce the level of indentation.

**Pull request merged**